pkg/util/kubernetes/apiserver/apiserver.go

```diff
@@ -98,13 +98,14 @@ func parseKubeConfig(fpath string) (*k8s.Config, error) {
 	return config, err
 }
 
-func (c *APIClient) GetComponents() *v1.ComponentStatusList {
-	componentsStatus, err := c.client.CoreV1().ListComponentStatuses(context.Background())
+func (c *APIClient) ComponentStatuses() (*v1.ComponentStatusList, error) {
+	ctx, cancel := context.WithTimeout(context.Background(), c.timeout)
+	defer cancel()
+	componentsStatus, err := c.client.CoreV1().ListComponentStatuses(ctx)
```

Comment:
you can simply `return c.client.CoreV1().ListComponentStatuses(ctx)` for now, and remove the rest of the method

Reply:
True, way more elegant.